.github/workflows/ci.yml

```
@@ -0,0 +1,290 @@
+  build:
+    runs-on: ubuntu-22.04
+    steps:
+      - name: ci/checkout-repo
+        uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c # v3.3.0
+      - name: ci/setup-node
+        uses: actions/setup-node@64ed1c7eab4cce3362f8c340dee64e5eaeef8f7c # v3.6.0
+        id: setup_node
+        with:
+          node-version-file: ".nvmrc"
+          cache: npm
+      - uses: actions/cache@2b8105bdae4b746666ee225970c4a281bbd9d51f # v3.2.4
+        id: npm-cache
+        with:
+          path: |
+            '**/node_modules'
+            'e2e/playwright/node_modules'
+            'e2e/cypress/node_modules'
+          key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}-${{ hashFiles('e2e/cypress/package-lock.json') }}-${{ hashFiles('e2e/playwright/package-lock.json') }}
+      - name: ci/get-node-modules
+        if: steps.npm-cache.outputs.cache-hit != 'true'
+        run: |
+          make node_modules
+          make e2e/playwright/node_modules
+      - name: ci/build
+        run: |
+          node --version
+          npm version
+          make dist
+      - name: ci/persist-artifact
+        uses: actions/upload-artifact@0b7f8abb1508181956e8e162db84b466c27e18ce # v3.1.2
+        with:
+          name: webapp-artifact
+          path: mattermost-webapp.tar.gz
+  upload-s3:
+    runs-on: ubuntu-22.04
+    needs: build
+    env:
+      REPO_NAME: ${{ github.event.repository.name }}
+    steps:
+    - name: ci/artifacts-downloads
+      uses: actions/download-artifact@e9ef242655d12993efdcda9058dee2db83a2cb9b  # v3.0.2
+      with:
+        name: webapp-artifact
+    - name: ci/aws-configure
+      uses: aws-actions/configure-aws-credentials@07c2f971bac433df982ccc261983ae443861db49 # v1-node16
+      with:
+        aws-region: ${{ secrets.AWS_REGION }}
+        aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+        aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+    # We need to sanitize the branch name before using it
+    - name: ci/sanitize-branch-name
+      id: branch
+      uses: transferwise/sanitize-branch-name@b10b4d524ac5a7b645b43a3527db3a6cca017b9d # v1
+    # Search for the string "pull" and replace it with "PR" in branch-name
+    - name: ci/sanitize-branch-name-replace-pull-with-PR-
+      run: echo "BRANCH_NAME=$(echo ${{ steps.branch.outputs.sanitized-branch-name }} | sed 's/^pull\//PR-/g')" >> $GITHUB_ENV
+    - name: ci/artifact-upload
+      run: |
+        aws s3 cp mattermost-webapp.tar.gz s3://${{secrets.AWS_S3_BUCKET_PRS}}/$REPO_NAME/$BRANCH_NAME/ --cache-control no-cache
+        aws s3 cp mattermost-webapp.tar.gz s3://${{secrets.AWS_S3_BUCKET_PRS}}/$REPO_NAME/commit/${{ github.sha }}/ --cache-control no-cache
+  prepare-docker-build:
+    runs-on: ubuntu-22.04
+    needs: upload-s3
+    # Run this Job, on every branch except "master","cloud-*","release-*" branches
+    if: |
+      ${{ !(github.ref_name == 'master' ||
+           startsWith(github.ref_name,'cloud-') ||
```

I actually just changed this one in the CircleCI config since I found out some people use branches named `cloud-`
```suggestion
           github.ref_name == 'cloud' ||
```

---

Proposed change is committed.